addons/pos_loyalty/static/src/overrides/models/loyalty.js

```diff
@@ -453,7 +456,27 @@ patch(Order.prototype, {
             }
             this.orderlines.remove(line);
         }
-        for (const claimedReward of productRewards.concat(otherRewards).concat(paymentRewards)) {
+        const allRewards = productRewards.concat(otherRewards).concat(paymentRewards);
+        var allRewardsMerged = [];
+        allRewards.forEach((reward) => {
+            if (reward.reward.reward_type == "discount") {
+                allRewardsMerged.push(reward);
+            }
+            else {
```

Comment:
You might wanna put this like `} else {` or it wont pass the linting test in forward ports I think